### PR TITLE
Add support for Space Duel’s cocktail mode.

### PR DESCRIPTION
**Pull request by ieure**
This change introduces a cabinet type setting which allows Space Duel to be placed into cocktail mode.  When run in a cocktail cabinet, the game:

 - Strobes the invert x & invert y outputs to rotate the video 180 degrees.  This is necessary because the monitor in the SD cocktail is mounted upside-down relative to the upright.
 - Draws the player 2 score in an orientation where they can be read by the second player, who stands on the opposite side of the cabinet as player one.

In addition to the driver support, I had to implement video flipping in avg_device, as it didn’t support it.

### CODE REVIEW

**Comment by ieure** [comment]
I'd appreciate some feedback on this change, as it's not quite right, and I'm not familiar enough with MAME to figure out why.  Everything *except* the actual video flipping is working.

I had to rework things a bit, since the AVG was set up in `bwidow()`, but it needs to be available in `spacduel_coin_counter_w()`.  Likely I'm missing something fairly obvious about how this is set up, but the prior art I found wasn't helpful.

File: src/mame/drivers/bwidow.cpp [hunk 1]
```diff
@@ -674,6 +676,14 @@ static INPUT_PORTS_START( spacduel )
 	/* bit 7 is tied to a 3kHz clock */
 	PORT_BIT( 0x80, IP_ACTIVE_HIGH, IPT_CUSTOM ) PORT_READ_LINE_MEMBER(bwidow_state, clock_r)
 
+	/* Cabinet type.  The cocktail's harness has a jumper between pins
+	   R and P, which maps to bit 7 of address 0x0907 */
+
```
Comment (happppp) [inline src/mame/drivers/bwidow.cpp:681]: You can use CONFNAME, CONFSETTING for things that are not DIP switches.

**Comment by happppp** [comment]
Once you get things working, don't forget to remove those printfs.

File: src/mame/video/avgdvg.cpp [hunk 2]
```diff
@@ -655,6 +655,11 @@ int avg_device::handler_7() // avg_strobe3
 
 	if (!OP0() && !OP2())
 	{
+        int x = m_xpos;
+		int y = m_ypos;
+
+		apply_flipping(x, y);
+
 		vg_add_point_buf(
```
[the remark this answers is not in the document]
Comment (dinkc64) [reply src/mame/video/avgdvg.cpp:663]: ieure, don't forget avg_common_strobe3
hap, void apply_flipping(int &x, int &y) const;

[the remark this answers is not in the document]
Comment (ieure) [reply src/mame/video/avgdvg.cpp:663]: Thank you, this solved my issue.

File: src/mame/drivers/bwidow.cpp (hunk 1 above)
Comment (ieure) [reply src/mame/drivers/bwidow.cpp:678]: Fixed!

**Comment by ieure** [comment]
Thank you for the suggestions, this should be fully working now.

File: src/mame/video/avgdvg.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (dinkc64) [reply src/mame/video/avgdvg.cpp:663]: excellent! :)

**Pull request merged**